### DISCUSSION TITLE
msbuild package restore not working in VS 2022 for .wixproj files

### DISCUSSION THREAD

**Discussion by ghost** [wixtoolset/issues#8805] [Questions]
Recently upgraded WIX via NuGet to the last 3.xxx (3.14.1) version. Now suddenly, msbuild won't restore the wix files under the packages directory. We use a command line like:
`msbuild.exe  -p:RestorePackagesConfig=true -t:Restore MyProject.sln /p:Configuration=Release /p:Platform="Any CPU"`

And we get the warning:
`The project file may be invalid or missing targets required for restore`

Anyone run into this issue?

**Comment by robmen** [comment]
I didn't think WiX v3.x supported the `Restore` target. The WiX v3 MSBuild targets were written years before this new style of doing restore came to MSBuild. In fact, tremendous effort went into re-writing the targets during WiX v4 to fully embrace to the new MSBuild/NuGet Restore model.

**Reply by ghost** [reply]
It used to work, though maybe it was by luck.

This was a working build script in 2019.

"C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\msbuild.exe"  -p:RestorePackagesConfig=true -t:Restore Services.sln
"C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\MSBuild\Current\Bin\msbuild.exe"  Services.sln /t:Clean,Build /p:Configuration=Release /p:Platform="Any CPU"

if NOT ["%errorlevel%"]==["0"] pause

Building in VS 2022 seemed to work fine, but maybe msbuild in 2022 doesn't support it.

I suppose I can try to upgrade to WiX 4.xxx, it hopefully isn't too bad since I'm not doing anything complicated.
